titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/iam/AwsIamUtil.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ *  * Copyright 2019 Netflix, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.netflix.titus.ext.aws.iam;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.titus.api.iam.model.IamRole;
+import com.netflix.titus.api.json.ObjectMappers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AwsIamUtil {
+    private static final Logger logger = LoggerFactory.getLogger(AwsIamUtil.class);
+
+    private static final String ASSUME_ACTION = "sts:AssumeRole";
+    private static final String ASSUME_EFFECT = "Allow";
+
+    /**
+     * Helper method to deserialize an IAM Role's assume policy and verify if
+     * another role can assume into it.
+     */
+    public static boolean canAssume(IamRole iamRole, String assumeResourceName) {
```

Comment:
Some unit tests would be great here. They would also server as a documentation for the IAM data parsed/processed here.

Reply:
Will add a test with some documentation of the policy format.